### DISCUSSION TITLE
Full body avatars for events

### DISCUSSION THREAD

**Discussion by arthurmougin** [mozilla/hubs#2718] [Avatar Questions] [answered]
Hello guys! 
for a big size event, I am looking to implement full-body avatars and optionally full-body tracking (inspired by VRCHAT). here are my questions about it :

1 - During the development phase, Should I try to build a custom client using the public default servers or do I need to buy my own AWS instance?

2 - do you know any best practices or advice on how to implement that feature correctly on Mozilla Hubs, I am quite new in WebXR programming?

3 - How could I allow full-body 3rd party avatars while yet limiting their size to make sure it does not pollute the experience for low-end users?

4 - Should I implement a fall-back solution for low-end hardware when an avatar is too difficult to render and how should I implement it?

For that event, I can accept losing the support of very low-end smartphones and computers if I cant implement a fallback solution...

Thank you in advance for your time and your answer, I am excited to implement that!

**Comment by misslivirose** [comment, marked as answer]
Sharing @netpro2k's answer from Discord: 

> First thing to figure out is if the vive trackers are even accessable in WebXR, if they are than this sounds possible, though it is going to be a lot of work. You should be able to do it with just a custom client I think, but everyone would need to be running that client. For development you can test against our dev server, but if you wanted to run it on more than locahlost you would need to set up a hubs cloud instance 

To echo what Dom said, you'll need to use Hubs Cloud if you want to host and distribute the custom client more widely - which sounds like might be the case for a big event. But for development purposes, building locally against our developer instance is fine. 

With regards to your questions 3 & 4, you could consider using the default Hubs avatars as a fallback, or design another low-poly, low-texture option. There is no "one right answer" as far as a possible design goes, but as far as I know, increasing the amount of physical movement data will have significant increases in performance on its own, as well keeping in mind polygon count and texture sizes for more detailed avatars. Some options could be to introduce a type of performance "level of detail" type system, or replacing expensive avatars with less detailed or less articulated ones on less powerful devices.

**Comment by arthurmougin** [comment]
To answer Dom: [they are accessible, it just depends on the browser](https://twitter.com/dmarcos/status/862459833613598720). 
Thank you for your answer, what you created as default avatars are a good starting point for my project. 

I guess I will have to dive into networked-aframe in to find how to implement that. 
If you say that it will work on your dev server, it means that I won't have to modify the network topology and I am really happy to know that!

This will be a big project indeed. I hope to share with you an update for this project soon on discord :)
